ls1intum/Artemis
# `Development`: Refactor participation access checks

### Checklist
#### General


#### Server

### Motivation and Context
While implementing another feature I needed to check access for participations.

### Description
Instead of either duplicating code, or making the `AuthCheckService` depend on programming exercise services which might not be available for all profiles, I added a new service that checks the authorization for participations.
This also simplifies the access checks for participations in other places.

The actual logic for the access checks has not changed, I only moved existing methods to the new service.

### Steps for Testing


Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with online editor enabled, 1 other text/fileUpload/modelling exercise

PR #6386 is based on top of this one, but the changes are somewhat independent of each other.
So you could test both at the same time by deploying #6386 and combining the testing steps.

1. As instructor, programming exercise: Check that you can see the results of the template/solution participation.
2. As instructor, both exercise types: Access student results for that exercise.
3. As student, both exercise types: Check that you access your own results.
4. As student, programming exercise: Check that the online editor works.

### Review Progress






#### Code Review
#### Manual Tests
- [x] Test 2 (access to participations also checked when testing #6386)

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| `ParticipationAuthorizationCheckService.java` | 100% | 100% |
